test/val/val_ext_inst_test.cpp

```diff
@@ -5204,6 +5204,20 @@ TEST_F(ValidateExtInst, GlslStd450InterpolateAtCentroidSuccess) {
   ASSERT_EQ(SPV_SUCCESS, ValidateInstructions());
 }
 
+TEST_F(ValidateExtInst, GlslStd450InterpolateAtCentroidInternalSuccess) {
```

Comment:
Could you add a test that makes sure this fails when "before_hlsl_legalization" is false?